### ISSUE TITLE
Investigate adding integration tests for injected header / footer

### ISSUE THREAD

**Issue by swirtSJW** [department-of-veterans-affairs/va.gov-cms#11253]
## Status
**[2024-05-21]** Revisit after the injected header flattening/separating work has been completed.

## User Story or Problem Statement

As maintainers of the injected header and footer, I would like to know when a PR breaks some portion of the injected header or footer so that we could prevent it from happening.

There's an existing [Cypress test created by Brooks Johnson](https://github.com/department-of-veterans-affairs/content-build/pull/1266) that asserts all web components are defined. This test might serve as a model for all or part of the work that would fall under this ticket. This test, though, would be dependent on #11260. If we do not decide to fetch all web components for the injected header/footer unconditionally, then that decision would affect how we write integration/e2e tests for this ticket.

## Information
From Swirt: Here is the most recent postmortem that mentions this issue https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/pull/913

## Acceptance Criteria
- [ ] Ticket(s) are created to test functionality of injected header such that a failure would prevent PR merge.
- [ ] If separate, ticket(s) are created to test functionality of injected footer such that a failure would prevent PR merge.

## Implementation steps
<optional>

## Design principles

Veteran-centered
- [ ] `Single source of truth`: Increase reliability and consistency of content on VA.gov by providing a single source of truth.
- [ ] `Accessible, plain language`: Provide guardrails and guidelines to ensure content quality.
- [ ] `Purposely structured content`: Ensure Content API can deliver content whose meaning matches its structure.
- [ ] `Content lifecycle governance`: Produce tools, processes and policies to maintain content quality throughout its lifecycle.

Editor-centered
- [ ] `Purpose-driven`: Create an opportunity to involve the editor community in VA’s mission and content strategy goals.
- [ ] `Efficient`: Remove distractions and create clear, straightforward paths to get the job done.
- [ ] `Approachable`: Offer friendly guidance over authoritative instruction.
- [ ] `Consistent`: Reduce user’s mental load by allowing them to fall back on pattern recognition to complete tasks.
- [ ] `Empowering`: Provide clear information to help editors make decisions about their work.

### CMS Team
Please check the team(s) that will do this work.

- [ ] `Program`
- [ ] `Platform CMS Team`
- [ ] `Sitewide Crew`
- [ ] `⭐️ Sitewide CMS`
- [x] `⭐️ Public Websites`
- [ ] `⭐️ Facilities`
- [ ] `⭐️ User support`

**Comment by kscoulter** [comment]
@department-of-veterans-affairs/vsa-public-websites-frontend please see the most recent postmortem that mentions this issue https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/pull/913

**Comment by FranECross** [comment]
@jilladams Is this still something that we should prioritize in an near-future sprint? Thanks!

**Comment by jilladams** [comment]
The associated epic here got punted a bit in light of the Federal Header Footer project, because Dave would ideally like to kill off the injected header altogether. (But as long as it lives, we should keep these tickets in the backlog, imo.)

**Comment by FranECross** [comment]
We don't really have a way to test TeamSites in CI. Will need to research.

**Comment by FranECross** [comment]
The injected header/footer is being separated from the 'main' header/footer, and there aren't / won't be any components in the injected header footer. Any needed tests are being written as part of the flattening work. Closing this ticket; if any additional work is needed in future, another ticket will be created.
